src/fu-engine.c

```diff
@@ -4335,6 +4335,13 @@ fu_engine_add_device (FuEngine *self, FuDevice *device)
 		fu_device_set_update_error (device, "VersionFormat is ambiguous for this device");
 	}
 
+	/* no vendor-id, and so no way to lock it down! */
+	if (fu_device_has_flag (device, FWUPD_DEVICE_FLAG_UPDATABLE) &&
+	    fu_device_get_vendor_id (device) == NULL) {
+		fu_device_remove_flag (device, FWUPD_DEVICE_FLAG_UPDATABLE);
+		fu_device_set_update_error (device, "Not updatable as no vendor ID set");
```

Comment:
I think the `Not updatable` prefix is implied by this coming from an `UpdateError` key.  Just saying `No vendor ID set` should probably be enough to convey the message

Reply:
Ack, fixed.